builtin/logical/database/rotation.go

```diff
@@ -64,6 +64,29 @@ func (b *databaseBackend) populateQueue(ctx context.Context, s logical.Storage)
 			continue
 		}
 
+		// If an account's NextVaultRotation period is nil, it means that the
+		// role was created before we added the `NextVaultRotation` field. In this
+		// case, we need to calculate the next rotation time based on the
+		// LastVaultRotation and the RotationPeriod. However, if the role was
+		// created with skip_import_rotation set, we need to use the current time
+		// instead of LastVaultRotation because LastVaultRotation is 0
+		if role.StaticAccount.NextVaultRotation.IsZero() {
+			log.Debug("NextVaultRotation is zero", roleName)
+			// Previously skipped import rotation roles had a LastVaultRotation value of zero
+			if role.StaticAccount.LastVaultRotation.IsZero() {
+				role.StaticAccount.SetNextVaultRotation(time.Now())
+			} else {
+				role.StaticAccount.SetNextVaultRotation(role.StaticAccount.LastVaultRotation)
+			}
+
+			entry, err := logical.StorageEntryJSON(databaseStaticRolePath+roleName, role)
+			if err != nil {
+				log.Warn("failed to build write storage entry", "error", err, "roleName", roleName)
+			} else if err := s.Put(ctx, entry); err != nil {
+				log.Warn("failed to write storage entry", "error", err, "roleName", roleName)
```

Comment:
Failure to write to the Vault storage is only a warning? Do we need better error handling here?

Reply:
Weird I thought I left a comment on this yesterday 😬 . I replaced this logic with the helper function [`StoreStaticRole`](https://github.com/hashicorp/vault/pull/30320/files#diff-eeb1be301e377e84d62e91cd8fe8170b31286490fd814c23a3436028a35a4aa1R81). I'm not sure what else we could do since this runs in a separate go routine?